src/index.js

```diff
@@ -3,51 +3,71 @@ import { createFilter } from '@rollup/pluginutils'
 export default function css(options = {}) {
   const filter = createFilter(options.include || ['**/*.css'], options.exclude)
   const styles = {}
-  const order = []
   let dest = options.output
-  let changes = 0
+  let hasChanged = false
+  let prevIds = []
 
   return {
     name: 'css',
+    buildStart() {
+      hasChanged = false
+    },
     transform(code, id) {
       if (!filter(id)) {
         return
       }
 
+      const { imports, codeWithoutImports } = splitImports(code);
+
       // When output is disabled, the stylesheet is exported as a string
       if (options.output === false) {
+        if (imports.length === 0) {
+          return {
+            code: `export default ['${JSON.stringify(code)}'`,
+            map: { mappings: '' }
+          }
+        }
+        const importNamed = imports.map((d, i) => `import i${i} from ${d}`).join('\n');
         return {
-          code: 'export default ' + JSON.stringify(code),
+          code: `
+            ${importNamed}
+            export default ${imports.map((_, i) => `i${i}`).join(' + ')} + '${JSON.stringify(codeWithoutImports)}'`,
```

Comment:
I think this would make more sense, but it needs tests to be sure:
```suggestion
            export default ${imports.map((_, i) => `i${i}`).join(' + ')}\n` + codeWithoutImports,
```

Reply:
fixed